### DISCUSSION TITLE
use layout only [Idea]

### DISCUSSION THREAD

**Discussion by hodev-dev** [tailwindlabs/tailwindcss#10725] [Ideas]
it would be nice if there is way to get all variation of layout  like`flex` and `grid` with all breakpoints without other styles ?
like only get flexbox and grid without text bg and other styles
like layout engine

**Comment by RobinMalfait** [comment]
Tailwind only generates what you are using. So if you are only using `flex` and `grid` related classes then only those will be generated.

**Reply by hodev-dev** [reply]
thank you for answer
tailwind works perfect as you mention but when we want to make  **dynamic layout base on data that come from server** we have to **make switch case of all possible width and height to make sure it is generated in compile time.** it would be great if we get static generator beside JIT to generate all the possible layout for dynamic usage

**Reply by hodev-dev** [reply]
for instance
an option to generate all flex,width,height without looking inside the content and purge the rest, and do default behavior for rest of classes

**Example** : inside ```tailwind.config``` would be good idea if we get option like below
```js
{
    "excludeJitClass": ["flex","width","height"]  // generate all variant of classes inside jit exlucde and use jit for rest of classess 
}
```
in my opinion with this approach we get best balance between size and usability
